plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/frontend.ts

```diff
@@ -337,6 +341,36 @@ const addToCartWithOptionsStore = store(
 				}
 			},
 		},
+		callbacks: {
+			setProductData: () => {
+				const { availableVariations, selectedAttributes } =
+					getContext< Context >();
```

Comment:
We can revisit this context later but sharing my thoughts here to not lose it.

I think we can even move availableVariations and selectedAttribute to Single Product Template and Single Product block under the `productData`, by doing so, we don't have to sync the state anymore.

Reply:
Does it sound good to revisit this in a follow-up? I'm a bit wary of adding too many things to the global store because once extensions start hooking into it, they might start depending on these data being always in the global store. Instead, if we keep it in the child blocks/stores, it's easier for us to introduce changes because other blocks can't rely on its existence.

Reply:
Yeah, I comment here to not forget about it. It's better to visit this topic later. This PR is too big already.